plugins/package-managers/maven/src/main/kotlin/tycho/TargetHandler.kt

```diff
@@ -34,48 +39,123 @@ import java.io.File
             val targetFiles = projectRoot.walkTopDown().filter {
                 it.name.endsWith(".target") && it.isFile
             }.toList()
 
-            return targetFiles.flatMapTo(mutableSetOf(), ::parseTargetFile)
+            return targetFiles.map(::parseTargetFile)
         }
 
         /**
          * Parse the given [targetFile] and extract the repository URLs referenced in it.
          */
-        private fun parseTargetFile(targetFile: File): Set<String> {
+        private fun parseTargetFile(targetFile: File): ParseTargetFileState {
             val handler = ElementHandler(ParseTargetFileState())
-                .handleElement("repository") { state, attributes ->
+                .handleElement("repository") { state, attributes, _ ->
                     state.repositoryUrls += attributes.getValue("location")
                     state
+                }.handleElement("groupId") { state, _, body ->
+                    state.addDependencyAttribute("groupId", body)
+                }.handleElement("artifactId") { state, _, body ->
+                    state.addDependencyAttribute("artifactId", body)
+                }.handleElement("classifier") { state, _, body ->
+                    state.addDependencyAttribute("classifier", body)
+                }.handleElement("type") { state, _, body ->
+                    state.addDependencyAttribute("packaging", body)
+                }.handleElement("version") { state, _, body ->
+                    state.addDependencyAttribute("version", body)
+                }.handleElement("dependency") { state, _, _ ->
+                    state.storeMavenDependency()
                 }
 
-            return parseXml(targetFile, handler).repositoryUrls
+            return parseXml(targetFile, handler)
         }
+
+        /**
+         * Generate an identifier for the given [artifact] that can be used in a mapping from Tycho artifacts to Maven
+         * dependencies.
+         */
+        private fun tychoId(artifact: Artifact): String = "${artifact.groupId}.${artifact.artifactId}"
     }
+
+    /**
+     * Try to map the given [tychoArtifact] to a Maven dependency based on dependency declarations found in the
+     * processed target files. In target files, it is possible to declare features that contain a number of regular
+     * Maven dependencies. There are also options to change the default resolution mechanism for such dependencies.
+     * If this is done, Tycho sometimes creates alternative identifiers for the dependencies. This function attempts
+     * to find the original Maven coordinates for affected artifacts. They are needed to retrieve the correct metadata.
+     * Result is *null* if no matching Maven dependency is found.
+     */
+    fun mapToMavenDependency(tychoArtifact: Artifact): Artifact? =
+        mavenDependencies[tychoArtifact.artifactId]?.also { art ->
```

Comment:
Nit: Maybe `dep` (for "dependency", like in `mavenDependencies`) instead of `art` for "artifact"?